### PR TITLE
Ionia manage card UI

### PR DESCRIPTION
**Pull request by godilite**

### CODE REVIEW

File: lib/src/screens/ionia/cards/ionia_buy_gift_card.dart [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+import 'package:flutter/services.dart';
+import 'package:keyboard_actions/keyboard_actions.dart';
+import 'package:cake_wallet/generated/i18n.dart';
+
+class IoniaBuyGiftCardPage extends BasePage {
+  IoniaBuyGiftCardPage()
+      : _amountFieldFocus = FocusNode(),
+        _amountController = TextEditingController();
+  @override
+  String get title => 'Enter Amount';
```
Comment (mkyq) [inline lib/src/screens/ionia/cards/ionia_buy_gift_card.dart:19]: Please add to localization file

**Review by mkyq — CHANGES_REQUESTED** [review]

File: lib/src/screens/dashboard/widgets/market_place_page.dart [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import 'package:cake_wallet/routes.dart';
+import 'package:cake_wallet/src/widgets/market_place_item.dart';
+import 'package:flutter/material.dart';
+import 'package:cake_wallet/generated/i18n.dart';
+
+class MarketPlacePage extends StatelessWidget {
+  final ScrollController _scrollController = ScrollController();
+  @override
```
[the remark this answers is not in the document]
Comment (godilite) [reply lib/src/screens/dashboard/widgets/market_place_page.dart:8]: done

File: lib/src/screens/ionia/cards/ionia_debit_card_page.dart [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+
+  String _formatPan(String pan) {
+    if (pan == null) return '';
+    return pan.replaceAllMapped(RegExp(r'.{4}'), (match) => '${match.group(0)}  ');
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final last4 = widget.isCardSample ? '0000' : widget.cardInfo.pan.substring(widget.cardInfo.pan.length - 5);
+    final spendLimit = widget.isCardSample ? '10000' : widget.cardInfo.spendLimit.toStringAsFixed(2);
```
Comment (mkyq) [inline lib/src/screens/ionia/cards/ionia_debit_card_page.dart:258]: I would to move it into getters

File: lib/src/screens/ionia/cards/ionia_debit_card_page.dart [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+                Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      'Expires',
+                      style: textXSmallSemiBold(),
+                    ),
+                    SizedBox(height: 4),
+                    Text(
+                      '${widget.cardInfo.expirationMonth ?? 'MM'}/${widget.cardInfo.expirationYear ?? 'YY'}',
```
Comment (mkyq) [inline lib/src/screens/ionia/cards/ionia_debit_card_page.dart:331]: Please move to localization file (MM and YY)

File: lib/src/screens/ionia/cards/ionia_manage_cards_page.dart [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    return TextField(
+      style: TextStyle(color: Colors.white),
+      decoration: InputDecoration(
+          filled: true,
+          contentPadding: EdgeInsets.only(
+            top: 10,
+            left: 10,
+          ),
+          fillColor: Colors.white.withOpacity(0.15),
+          hintText: 'Search',
```
Comment (mkyq) [inline lib/src/screens/ionia/cards/ionia_manage_cards_page.dart:197]: Please add to localization file

**Review by mkyq — APPROVED** [review]

**Pull request merged**